angular/src/lib/select/select.component.ts

```diff
@@ -115,306 +226,474 @@ export class SelectComponent implements OnInit, AfterContentChecked, AfterConten
+      }
+    }
+    return selectItems;
   }
 
-  get buttonClasses (): string {
-    return (
-      `${(this.buttonClass && `${this.buttonClass}`) || ''}` +
-      `${(this.disabled && ` disabled`) || ''}`
-    );
+
+  updateSelectedItem(item: any): void {
+    this.selectedItem = this.findSelectOption(item, this.selectOptionsToDisplay);
+    this.selectedItemUpdated = true;
   }
 
-  /** The currently selected option. */
-  get selected(): ListItemComponent | ListItemComponent[] {
-    return this.isMulti ? this._selectionModel.selected : this._selectionModel.selected[0];
+  findSelectOption(item: any, options: any[]) {
+    const index: number = this.findSelectOptionIndex(item, options);
+    return (index !== -1) ? options[index] : null;
   }
 
-  /** Toggles the overlay panel open or closed. */
-  toggle = (): void => {
-    this.overlayOpen ? this.close() : this.open();
-    this.focus();
+  findSelectOptionIndex(item: any, options: any[]): number {
+    let index: number = -1;
+    if (options) {
+      for (let i = 0; i < options.length; i++) {
+        if ((item === null && options[i].value === null) || item === options[i].value) {
+          index = i;
+          break;
+        }
+      }
+    }
+    return index;
   }
 
-  open = (): void => {
-    if (this.disabled || !this.selectOptions || !this.selectOptions.length || this.overlayOpen) {
-      return;
+  onFilter(event): void {
+    const inputValue = event.target.value;
+    if (inputValue && inputValue.length) {
+      this.filterValue = inputValue;
+      this.startFilter();
+    } else {
+      this.filterValue = null;
+      this.selectOptionsToDisplay = this.options;
     }
-    this.overlayOpen = true;
   }
 
-  close = (): void => {
-    if (this.overlayOpen) {
-      this.overlayOpen = false;
+  startFilter() {
+    const filterBy: string[] = this.filterBy.split(',');
+
+    if (this.options && this.options.length) {
+      this.selectOptionsToDisplay = FilterUtils.filter(this.options, filterBy, this.filterValue, this.filterMode);
     }
   }
 
-  /** Focuses the origin button of the select component. */
-  focus = (): void => {
-    this.originButton.el.nativeElement.focus();
+  resetFilter(): void {
+    if (this.filterViewChild && this.filterViewChild.nativeElement) {
+      this.filterValue = null;
+      this.filterViewChild.nativeElement.value = '';
+    }
+    this.selectOptionsToDisplay = this.options;
   }
 
-  currentValue = () => {
-    if (!this._selectionModel) { return; }
+  onSelectItemClick(event) {
+    const option = event.option;
 
-    const selected = this._selectionModel.selected;
-    if (!this.isMulti && selected.length) {
-      return selected[0].label; }
-    if (selected.length === 1) {
-      return `${selected.length} Item Selected`;
-    } else if (selected.length) {
-      return `${selected.length} Items Selected`;
+    if (!option.disabled) {
+      this.selectItem(event, option);
     }
+    setTimeout(() => {
```

Comment:
Should this timeout value be configurable?

Reply:
Yeah good idea, it can be, i can add that.